packages/screens/RWA/components/RWAScreenContainer/RWASidebar/RWASidebar.tsx

```diff
@@ -0,0 +1,233 @@
+import { Separator } from "../../../../../components/separators/Separator";
+import { SpacerColumn } from "../../../../../components/spacer";
+import { useRWASideBar } from "../../../../../context/SidebarProvider";
+import { useIsMobile } from "../../../../../hooks/useIsMobile";
+import { useIsLightTheme, useTheme } from "../../../../../hooks/useTheme";
+import { useAppNavigation } from "../../../../../utils/navigation";
+import {
+  MOBILE_HEADER_HEIGHT,
+  MOBILE_SIDEBAR_MAX_WIDTH,
+  fullSidebarWidth,
+  headerHeight,
+  layout,
+  smallSidebarWidth,
+} from "../../../../../utils/style/layout";
+
+const RWATopLogo = () => {
+  const navigation = useAppNavigation();
+  const isLightTheme = useIsLightTheme();
+
+  return (
+    <View style={{ flex: 1, justifyContent: "center" }}>
+      <TouchableOpacity
+        style={{ marginHorizontal: layout.spacing_x0_5 }}
+        onPress={() => navigation.navigate("RWAHome")}
+      >
+        <SVG
+          width={68}
+          height={68}
+          source={isLightTheme ? RWALightLogo : RWADarkLogo}
+        />
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const SpringConfig: WithSpringConfig = {
+  stiffness: 100,
+  mass: 0.5,
+  restDisplacementThreshold: 0.2,
+};
+
+const SidebarSeparator: React.FC = () => {
+  const theme = useTheme();
+  return (
+    <View
+      style={{
+        height: 1,
+        marginHorizontal: layout.spacing_x2,
+        backgroundColor: theme.borderColor,
+        marginBottom: layout.spacing_x1,
+      }}
+    />
+  );
+};
+
+export const RWASidebarMobile: React.FC = () => {
+  const { height: windowHeight, width: windowWidth } = useWindowDimensions();
+  // const navigation = useAppNavigation();
+  const { isSidebarExpanded, dynamicSidebar } = useRWASideBar();
+  const theme = useTheme();
+
+  const layoutStyle = useAnimatedStyle(
+    () => ({
+      width: isSidebarExpanded
+        ? withSpring(
+            windowWidth < MOBILE_SIDEBAR_MAX_WIDTH
+              ? windowWidth
+              : MOBILE_SIDEBAR_MAX_WIDTH,
+            SpringConfig,
+          )
+        : withSpring(0, SpringConfig),
+    }),
+    [isSidebarExpanded, windowWidth],
+  );
+
+  const onRouteChange = (name: SidebarType["route"]) => {
+    // @ts-expect-error
+    navigation.navigate(name);
+  };
+
+  return (
+    <Animated.View
+      style={[
+        SideBarMobileContainerCStyle,
+        layoutStyle,
+        {
+          height: windowHeight - MOBILE_HEADER_HEIGHT,
+          backgroundColor: theme.headerBackgroundColor,
+          borderColor: theme.borderColor,
+        },
+        !isSidebarExpanded && { borderRightWidth: 0 },
+      ]}
+    >
+      <FlatList
+        showsVerticalScrollIndicator={false}
+        data={Object.values(dynamicSidebar)}
+        keyExtractor={(item) => item.id}
+        renderItem={({ item }) => (
+          <RWASidebarButton key={item.id} onPress={onRouteChange} {...item} />
+        )}
```

Comment:
You can delete `RWASidebarButton.tsx` and use the existing `SidebarButton` here IMO
```suggestion
        renderItem={({ item }) => (
          <SidebarButton onPress={onRouteChange} {...item} />
        )}
```

Reply:
It's outdated because i change naming here, but idk why you have this version.. because i have nothing to push and im up to date with the branch..
But overall this is not the same UI.. so i can't use the existant one

Reply:
Sorry, I forgot that it will be in another repo